benchmarks/sort/sort_benchmark.cpp

```diff
@@ -0,0 +1,580 @@
+// Defining _CubLog here allows us to avoid that code path, however disabling
+// some debugging diagnostics.
+//
+#    define _CubLog
+#    include <thrust/device_ptr.h>
+#    include <thrust/sort.h>
+#  else // #if (KOKKOS_COMPILER_CLANG < 900)
+#    include <thrust/device_ptr.h>
+#    include <thrust/sort.h>
+#  endif // #if (KOKKOS_COMPILER_CLANG < 900)
+#endif   // #if defined(KOKKOS_ENABLE_CUDA)
+// clang-format on
+
+template <typename ViewType>
+void iota(ViewType view)
+{
+  using ExecutionSpace = typename ViewType::execution_space;
+  auto const n = view.extent(0);
+  Kokkos::parallel_for("iota", Kokkos::RangePolicy<ExecutionSpace>(0, n),
+                       KOKKOS_LAMBDA(int i) { view(i) = i; });
+}
+
+template <typename ValueType, typename DeviceType, typename SizeType>
+struct SortKokkos
+{
+  using value_type = ValueType;
+  using device_type = DeviceType;
+
+  static Kokkos::View<SizeType *, DeviceType>
+  sortAndComputePermutation(Kokkos::View<ValueType *, DeviceType> view)
+  {
+    using ViewType = Kokkos::View<ValueType *, DeviceType>;
+    using ExecutionSpace = typename ViewType::execution_space;
+    using CompType = Kokkos::BinOp1D<ViewType>;
+
+    int const n = view.extent(0);
+
+    Kokkos::MinMaxScalar<ValueType> result;
+    Kokkos::MinMax<ValueType> reducer(result);
+    Kokkos::parallel_reduce(
+        "find_min_max_view", Kokkos::RangePolicy<ExecutionSpace>(0, n),
+        Kokkos::Impl::min_max_functor<ViewType>(view), reducer);
+
+    Kokkos::BinSort<ViewType, CompType, typename ViewType::device_type,
+                    SizeType>
+        bin_sort(view, CompType(n / 2, result.min_val, result.max_val), true);
+    bin_sort.create_permute_vector();
+    bin_sort.sort(view);
+
+    return bin_sort.get_permute_vector();
+  }
+
+  static void sort(Kokkos::View<ValueType *, DeviceType> view)
+  {
+    auto permute = sortAndComputePermutation(view);
+    std::ignore = permute;
+  }
+
+  static void applyPermutation(Kokkos::View<SizeType *, DeviceType> permute,
+                               Kokkos::View<ValueType *, DeviceType> in,
+                               Kokkos::View<ValueType *, DeviceType> &out)
+  {
+    using ExecutionSpace = typename DeviceType::execution_space;
+    int const n = in.extent(0);
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<ExecutionSpace>(0, n),
+        KOKKOS_LAMBDA(int const i) { out(permute(i)) = in(i); });
+  }
+};
+
+#if defined(KOKKOS_ENABLE_SERIAL)
+template <typename ValueType, typename DeviceType, typename SizeType>
+struct SortStd
+{
+  using value_type = ValueType;
+  using device_type = DeviceType;
+
+  static void sort(Kokkos::View<ValueType *, DeviceType> view)
+  {
+    static_assert(std::is_same<Kokkos::Serial,
+                               typename DeviceType::execution_space>::value,
+                  "");
+
+    std::sort(view.data(), view.data() + view.extent(0));
+  }
+
+  static auto computePermutation(Kokkos::View<ValueType *, DeviceType> view)
```

Comment:
`auto` without trailing return type is C++14.  You need to set the standard with CMake

Reply:
Fixed.